src/Dialect/ONNX/Rewrite.cpp

```diff
@@ -209,6 +209,61 @@ bool haveSameStaticShape(Value lhs, Value rhs) {
   return hasStaticShape(lhsT) && (getShape(lhsT) == getShape(rhsT));
 }
 
+// Match v = shape_transform(X*A + B).
```

Comment:
Do we cover the case where instead of `X*A+B` we have a `Gemm` op?

Reply:
Good idea. I will add Gemm.

Reply:
I added the case for Gemm.